### PR TITLE
Bump `umbral-pre` to `0.10.0`

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:** 
-  1

**What this does:**
- Bumps `rust-umbral` version

**Why it's needed:**
- `0.10.0` contains a [fix for compatibility issue](https://github.com/nucypher/rust-umbral/pull/122) with `ferveo`
- We need it to release a new `nucypher-core` version with `ferveo` support

### CODE REVIEW

**Review by derekpierre — COMMENTED** [review]
Is this a reason to bump the version of `nucypher-core`? i.e. do we need to do a new nucypher-core release, or we are still in the process of testing changes?
 
As a side note, I intend to bump the release version for PR #54 .

**Comment by piotr-roslaniec** [comment]
I think we're still in the process of testing. Once we have `nucypher-ts` and `nucypher` working end-to-end, I think it would be a good time to release `nucypher-core`. We can always release an alpha version if we want to make `nucypher` and `nucypher-ts` integration a bit easier.

**Review by derekpierre — APPROVED** [review]
🎸

**Comment by derekpierre** [comment]
> I think we're still in the process of testing. Once we have `nucypher-ts` and `nucypher` working end-to-end, I think it would be a good time to release `nucypher-core`. We can always release an alpha version if we want to make `nucypher` and `nucypher-ts` integration a bit easier.

Wait. Thinking about this more...

Do you intend to merge this change before the testing is completed, OR are you testing currently and will update this PR as needed?

**Comment by piotr-roslaniec** [comment]
From my perspective, I don't see any additional changes in `rust-umbral` I could need. I only need to resolve certain dependency conflicts. You can see in `Cargo.toml` that this change was already tested prior to `0.10.0` release.

**Comment by derekpierre** [comment]
> From my perspective, I don't see any additional changes in `rust-umbral` I could need. I only need to resolve certain dependency conflicts. You can see in `Cargo.toml` that this change was already tested prior to `0.10.0` release.

The concern I have is, do these changes have downstream effects on `nucypher/nucypher` and are those being done in tandem? If there are downstream effects, we should have those as a PR in `nucypher/nucypher` before actually merging this change.

**Review by KPrasch — APPROVED** [review]

**Comment by piotr-roslaniec** [comment]
Let's postpone merging until I sort out the downstream changes (if needed).

**Comment by piotr-roslaniec** [comment]
I think we can merge this after successful merge of https://github.com/nucypher/nucypher/pull/3135 and merge of downstream changes in other PRs

**Comment by derekpierre** [comment]
> I think we can merge this after successful merge of [nucypher/nucypher#3135](https://github.com/nucypher/nucypher/pull/3135) and merge of downstream changes in other PRs

I'm confused about the ordering. Doesn't this PR need to be merged/released first and then 3135 updated to depend on it?

**Comment by derekpierre** [comment]
> Sorry, let me clarify:
> 
> 1. #3155 is a canary for testing changes in this PR (and other PRs)
> 2. If #3155 is merged, then it would mean this PR checks out
> 3. At some point we'll release a new version of `nucypher-core` and introduce it to `nucypher`, and replace dependency on the pre-release package (in `nucypher/requirements.txt`) with a proper `nucypher-core` package.

Yep, but I don't think 3135 should be merged first. It should be used for testing no doubt, but merging when using a forked/branch github dependency on `nucypher-core` can cause issues.

I think the options are:

1. merge the `nucypher-core` changes, do a release, then update 3135 to depend on that release. This is what I thought we've been doing until now - but if we want to change that to reduce `nucypher-core` releases that's fine

OR

2. If we want to collect various changes in `nucypher-core` before a release is actually done, merge the `nucypher-core` change first and have the `nucypher` PR update the dependency to the `nucypher-core:main` commit in the github repos (and not a forked/branched version). If we are going to do this then the Pipfile should be changed as well to make that clear and not just the requirements.txt/dev-requirements.txt

cc @KPrasch in case he has any other ideas.

**Comment by piotr-roslaniec** [comment]
I'd prefer option 2). And yes, keeping a fork in `nucypher` dependencies is not really an option.

**Review by fjarri — APPROVED** [review]

**Pull request merged**
